client.go

```diff
@@ -779,3 +798,29 @@ func (c *Client) StandardClient() *http.Client {
 		Transport: &RoundTripper{Client: c},
 	}
 }
+
+var (
+	_ HTTPClientFactory = &CleanPooledClientFactory{}
+	_ HTTPClient        = &idleConnectionsClosingClient{}
+)
+
+type CleanPooledClientFactory struct {
+}
+
+func (f *CleanPooledClientFactory) New() HTTPClient {
```

Comment:
Something to consider is adding an `error` to the returned values. Some factories may have a problem while constructing a client and they will not have a good way to propagate the error up. WDYT?